build/v1/types.go

```diff
@@ -1022,23 +1026,37 @@ type ImageChangeTrigger struct {
 	Paused bool `json:"paused,omitempty" protobuf:"varint,3,opt,name=paused"`
 }
 
+// BuildImageChangeTriggerKey captures the required elements for identifying the ImageStreamTag referenced by the more
+// generic ObjectReference BuildTriggerPolicy.ImageChange.From.  It is used by ImageChangeTriggerStatus, where a
+// specific instance of ImageChangeTriggerStatus in maintained in BuildConfigStatus.ImageChangeTriggers for each entry
+// in the BuildConfigSpec.Triggers array where the BuildTriggerPolicy.ImageChange pointer is set to a non-nil value
+type BuildImageChangeTriggerKey struct {
```

Comment:
`ImageStreamTagReference`?

It's legal to have BuildConfig in ns/one triggered by an imagestreamtag in ns/two?

Reply:
> It's legal to have BuildConfig in ns/one triggered by an imagestreamtag in ns/two?

yes.  common, even.  "i want my app image in my NS to be rebuilt every time the corp base image in some shared NS it is built on top of is updated"

Reply:
Changed to `ImageStreamTagReference` in next push